### PR TITLE
Ministry codes

### PR DESCRIPTION
**Pull request by sangster**
This PR completes the most recent action-items for [#128093147 Add Ministry Model (Ministry_name, description)](https://www.pivotaltracker.com/story/show/128093147).

It removes the unneeded `description` attribute (minor) and adds the `code` attribute (major) to the `Ministry` model.

## Ministry Codes

Every ministry now requires a `code` to be present. This code identifies the affiliation of the ministry and helps determine who gets billed for what. There is basically a hard-coded list of codes to choose from, specified in the `Ministry` class.

The part that makes this interesting is that the codes are organized into a 3-tiered hierarchy. Instead of selecting from a large flat-list of codes, they want to drill down through the hierarchy. The requested workflow is to choose a top-level category, then the sub-category, then finally selecting the code from the sub-category.

We use progressive enhancement to improve the UI for the Ministry form. The page first loads with a standard `<select>` element to choose the code (from a flat-list). At this point, we replace this selector with a multi-level select component, provided by the `jQuery.dropdown` plugin.

### Notes
 * The bulk of this PR is the `app/assets/javascripts/ministry/code.coffee.erb` file, which enhances the code selector.
 * The `jQuery.dropdown` plugin has been added to the `vender/` directory. Unless you're bored, you shouldn't feel compelled to review that third-party code. It's not amazing, but it works.
 * The CSS choices made in `app/assets/stylesheets/widgets/dropdown-variables.scss` are solely to make the UI provided by `jQuery.dropdown` to look as similar as possible to the UI provided by the `jQuery.chosen` plugin.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -28,29 +28,6 @@
   add_index "active_admin_comments", ["namespace"], name: "index_active_admin_comments_on_namespace"
   add_index "active_admin_comments", ["resource_type", "resource_id"], name: "index_active_admin_comments_on_resource_type_and_resource_id"
 
-  create_table "audits", force: :cascade do |t|
```
Comment (mcwrightballistiq) [inline db/schema.rb]: Why is this table being removed in this PR?

File: app/admin/ministry.rb [hunk 2]
```diff
@@ -34,7 +33,7 @@
     f.semantic_errors
     f.inputs do
       f.input :name
-      f.input :description, as: :ckeditor
```
Comment (mcwrightballistiq) [inline app/admin/ministry.rb]: I did a quick search and don't see the CKeditor in use anywhere else in this project please confirm and remove the 3rd party lib if so.

Comment (sangster) [inline app/admin/ministry.rb]: The CK Editor is still used for editing other model attributes, like `Course#description` and `Conference#description`.

File: db/schema.rb (hunk 1 above)
Comment (mcwrightballistiq) [inline db/schema.rb]: Ok, Thanks for the explanation.

File: app/admin/ministry.rb (hunk 2 above)
Comment (mcwrightballistiq) [inline app/admin/ministry.rb]: Thanks for confirming. In my quick skim I had just seen the attribute 'description' and assumed it was one and the same.   Thank you for confirming.

**Pull request merged**